RELEASE.md

```diff
@@ -210,7 +210,7 @@ An example of a previous post: https://www.reddit.com/r/podman/comments/10moat6/
 #### Email announcement
 
 1. Send email to devtools-team at redhat.com, Podman-Desktop at redhat.com
-1. Send email to product-announce at redhat.com (high level)
+1. Check in with the PM about the product-announce announcement (high level) and the Google form https://docs.google.com/forms/d/e/1FAIpQLSeGMs3Okk0ccxoaafN5SPasf0pXeCtS7iQqNFkaT2sda4VpVw/viewform
```

Comment:
I'm not sure we should have Red Hat internal links here.

@slemeur WDYT ?

Reply:
I'm not Stevan, but we should not. :)

Reply:
@deboer-tim @jeffmaury fixed!